pkg/pub_integration/lib/script/publisher.dart

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:meta/meta.dart';
+
+import '../src/pub_http_client.dart';
+import '../src/pub_tool_client.dart';
+import '../src/test_data.dart';
+
+typedef InviteCompleterFn = Future Function();
+
+/// A single object to execute publisher-related integration script and
+/// verification tests with the `pub` tool on the fake site.
+class PublisherScript {
+  final String pubHostedUrl;
+  final String credentialsFileContent;
+  PubHttpClient _pubHttpClient;
+  PubToolClient _pubToolClient;
+
+  Directory _temp;
+
+  PublisherScript({
+    @required this.pubHostedUrl,
+    @required this.credentialsFileContent,
+  });
+
+  /// Verify all integration steps.
+  Future verify() async {
+    assert(_pubHttpClient == null);
+    assert(_pubToolClient == null);
+    _pubHttpClient = PubHttpClient(pubHostedUrl);
+    _temp = await Directory.systemTemp.createTemp('pub-integration');
+    try {
+      _pubToolClient = await PubToolClient.create(
+          pubHostedUrl: pubHostedUrl,
+          credentialsFileContent: credentialsFileContent);
+
+      await _publishVersion('1.0.0');
+      await Future.delayed(Duration(seconds: 1));
+      await _verifyDummyPkg(version: '1.0.0');
+
+      await _pubHttpClient.createPublisher(
+        authToken: 'user-at-example-dot-com',
+        publisherId: 'example.com',
+        accessToken: 'access-at-example-dot-com',
+      );
+      await _pubHttpClient.setPackagePublisher(
+        authToken: 'user-at-example-dot-com',
+        package: '_dummy_pkg',
+        publisherId: 'example.com',
+      );
+
+      final publisherInfo =
+          await _pubHttpClient.getContent('/api/packages/_dummy_pkg/publisher');
+      if (publisherInfo != '{"publisherId":"example.com"}') {
```

Comment:
Never assume how JSON is rendered... I don't think our API attempts to make any such promises.